Source/UnitEx.cpp

```diff
@@ -1175,8 +1186,13 @@ int UnitEx::GetSurveyedBy()
 		return HFLNOTINITED;
 	}
 
-	BeaconData* p = (BeaconData*)((int)(*unitArray) + (unitID * 120) + 0x58);
-	return p->surveyedBy;
+	return GetBeaconData(unitID).surveyedBy;
+}
+
+// HFL initialization must be checked before calling GetLabData
+LabData& GetLabData(int unitID) 

```

Comment:
This could be switched to returning a pointer if preferred.

Reply:
I think I actually prefer the reference, as you've already done here. A pointer is nullable. A reference isn't. I would not expect this method to ever return `nullptr` (for a Lab unit), so I think reference syntax makes more sense here.

For a non-Lab unit, this method should probably `throw`. So again, no need to ever return `nullptr`.

I also find the reference syntax looks a bit cleaner than pointer syntax.